cardano-api/src/Cardano/Api/Types.hs

```diff
@@ -110,8 +136,9 @@ data Address
 
 
 data Certificate
-  = ByronCertificate
-  | ShelleyDelegationCertificate !ShelleyDelegationCertificate
+  = ShelleyDelegationCertificate !ShelleyCertificate
+  | ShelleyStakePoolCertificate !ShelleyCertificate
+  deriving (Eq, Show)
```

Comment:
If these are the same type, do we need two constructors?

Reply:
I use them in the CBOR: https://github.com/input-output-hk/cardano-node/pull/890/commits/68ef28cc0b6e0000820d5537417e82197b46ad91#diff-14c30ebd1606eb8a1efa6f0e506e71a8R79